$OpenBSD: patch-ed_refresh_c,v 1.1 2015/01/12 13:10:11 millert Exp $

Fix out of bounds crash in emacs mode when erasing long lines.
See http://bugs.gw.com/view.php?id=412

--- ed.refresh.c.orig	Tue Jan  6 14:30:14 2015
+++ ed.refresh.c	Tue Jan  6 14:26:58 2015
@@ -1299,7 +1299,7 @@ ClearDisp(void)
     CursorV = 0;		/* clear the display buffer */
     CursorH = 0;
     for (i = 0; i < TermV; i++)
-	(void) memset(Display[i], 0, TermH * sizeof(Display[0][0]));
+	(void) memset(Display[i], 0, (TermH + 1) * sizeof(Display[0][0]));
     OldvcV = 0;
     litlen = 0;
 }
